s3-handler/main.go

```diff
@@ -131,13 +151,55 @@ func Handler(request events.S3Event) (Response, error) {
 	}, nil
 }
 
+func parseSampleRateRules(input []byte) ([]sampleRateRule, error) {
+	var result []sampleRateRule
+	err := json.Unmarshal(input, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, rule := range result {
+		if rule.SampleRate == 0 {
+			rule.SampleRate = 1
+		}
+	}
```

Comment:
This doesn't work because `rule` is a copy here. You need to use `result[i].SampleRate = 1`

https://go.dev/play/p/NTu8q-o5xPG